### PR TITLE
LESQ-693 -  Refactor: share link to pupil experience by subject

### PR DESCRIPTION
**Pull request by k-huggs**
## Description

Music year: 2001

- Amends how links are generated to be by subject 

## Issue(s)

Fixes #LESQ-693

## How to test

1. Go to https://deploy-preview-2304--oak-web-application.netlify.thenational.academy
2. Click on \_\_\_\_\_\_
3. You should see \_\_\_\_\_\_

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/components/TeacherViews/LessonShare/LessonShare.view.tsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ const classroomActivityMap: Partial<
 };
 
 // Temporary - list of lessons live on pupil experience for sharing
```
Comment (njblore) [inline src/components/TeacherViews/LessonShare/LessonShare.view.tsx:67]: these comments just need updating to say subjects now

**Review by njblore — CHANGES_REQUESTED** [review]

**Review by njblore — APPROVED** [review]

**Review by jack-skerrett-oak — CHANGES_REQUESTED** [review]
I didn't check the ticket before it got worked on! 

We are also going to need to extend this to the weird, psuedo-english subjects in KS2 (english grammar, english reading for pleasure, english spelling)

I need to pop into a meeting but will continue having a look after!

**Comment by k-huggs** [comment]
> I didn't check the ticket before it got worked on!
> 
> We are also going to need to extend this to the weird, psuedo-english subjects in KS2 (english grammar, english reading for pleasure, english spelling)
> 
> I need to pop into a meeting but will continue having a look after!

Added ✅

**Comment by rightsaidjames** [comment]
I'm waiting to hear back from @aakeshp on the priority of this, but me and @jack-skerrett-oak have discovered a small flaw in the current share URL generation logic that affects the 'weird, psuedo-english subjects in KS2'.

This share page from the English Spelling programme:
https://deploy-preview-2304--oak-web-application.netlify.thenational.academy/teachers/programmes/english-spelling-primary-ks2-l/units/silent-letters-3eb2/lessons/to-investigate-silent-letters-c9gk4r/share?preselected=all 

... links out to:

 https://thenational.academy/pupils/lessons/to-investigate-silent-letters-c9gk4r?share=true
 
 .... but this lesson's 'View all lessons' link points to https://classroom.thenational.academy/units/the-day-the-crayons-quit-persuasive-writing-0c4a because the lesson exists in both units.
 
If the share URL was:
https://www.thenational.academy/pupils/l/silent-letters-3eb2/lessons/to-investigate-silent-letters-c9gk4r?share=true (i.e. it included the `/l/` prefix followed by the unit slug and another `/`)

.... then the 'View all lessons' link correctly points to:
 https://classroom.thenational.academy/units/silent-letters-3eb2
 
 So... @k-huggs / @njblore I think we need to fix this by adopting the same URL structure that's used in classroom.thenational.academy for all pupil experience share links:
 
` /pupils/l/[unitSlug]/lessons/[lessonSlug]`

You can see this in action here:
https://classroom.thenational.academy/units/the-canterbury-tales-the-knights-tale-chaucer-e7e8

And here's the Acorn code that enables it:
https://github.com/oaknational/Acorn/pull/770/files#diff-7c30b00d23f898eb7147c7db64e229215da2171ee958161c636620a1e1765e0eR156

Is this fix doable, and if so is this something that @simonh-oak and/or @benprotheroe could help with?

**Review by jack-skerrett-oak — APPROVED** [review]

**Review by jack-skerrett-oak — COMMENTED** [review]
Should have said in that approval, we are not requiring any more changes on this PR. We have created a ticket to address the issue of lessons in multiple units here: https://www.notion.so/oaknationalacademy/Sharing-lessons-to-pupils-should-not-direct-them-to-canonical-pupil-pages-5b1560b32b5f46b39aaf3277738f1ae8?pvs=4

**Pull request merged**